src/test/java/seedu/address/logic/commands/PinCommandTest.java

```diff
@@ -0,0 +1,82 @@
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandFailure;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_PERSON;
+import static seedu.address.testutil.TypicalIndexes.INDEX_SECOND_PERSON;
+import static seedu.address.testutil.TypicalPersons.getTypicalAddressBook;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.person.Person;
+
+
+/**
+ * Contains integration tests (interaction with the Model) for {@code PinCommand}.
+ */
+public class PinCommandTest {
+    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+
+    @Test
+    public void execute_validIndexUnfilteredList_success() {
+        Person personToPin = model.getFilteredPersonList().get(INDEX_FIRST_PERSON.getZeroBased());
+        PinCommand pinCommand = new PinCommand(INDEX_FIRST_PERSON);
+
+        String expectedMessage = String.format(PinCommand.MESSAGE_PIN_PERSON_SUCCESS,
+                Messages.formatShort(personToPin));
+
+        ModelManager expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
+        expectedModel.addPinnedPersonList(personToPin);
+
+        assertCommandSuccess(pinCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_invalidIndexUnfilteredList_throwsCommandException() {
+        Index outOfBoundIndex = Index.fromOneBased(model.getFilteredPersonList().size() + 1);
+        PinCommand pinCommand = new PinCommand(outOfBoundIndex);
+
+        assertCommandFailure(pinCommand, model, Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void execute_duplicatePerson_throwsCommandException() {
+        Person validPerson = model.getFilteredPersonList().get(INDEX_FIRST_PERSON.getZeroBased());
+        PinCommand pinCommand = new PinCommand(INDEX_FIRST_PERSON);
+        ModelManager filledModel = new ModelManager(model.getAddressBook(), new UserPrefs());
+        filledModel.addPinnedPersonList(validPerson);
+
+        assertThrows(CommandException.class,
+                PinCommand.MESSAGE_DUPLICATE_PERSON, () -> pinCommand.execute(filledModel));
+    }
+
+
+    @Test
+    public void equals() {
+        PinCommand pinFirstCommand = new PinCommand(INDEX_FIRST_PERSON);
+        PinCommand pinSecondCommand = new PinCommand(INDEX_SECOND_PERSON);
+
+        assertTrue(pinFirstCommand.equals(pinFirstCommand));
```

Comment:
I would suggest to make an exact copy of the command and check for equality. Currently, if `equals()` is implemented by reference equality this test will pass:
```java
    @Override
    public boolean equals(Object other) {
        return other == this;
    }
```

Though, to be fair some other tests will fail.

Reply:
I think its fine to leave that test case in, as it is meant to cover this path 
![image](https://github.com/user-attachments/assets/ebfec64c-2ee3-47fb-88a9-dfdcdc6714c9)
whereas your suggestion would be testing this path
![image](https://github.com/user-attachments/assets/7282e85e-04b4-4984-ab92-eb81b1f5a5ec)

Reply:
hmm... technically ```assertFalse(pinFirstCommand.equals(pinSecondCommand));``` does cover that path, just the negative match of it. But i guess i can add one for the positive as well.